marketplaceItems/drawApp/fingerDraw_app.js

```diff
@@ -0,0 +1,596 @@
+                setUpDesktopMode();
+            }
+        }
+    }
+
+    /* GET ANIMATION DATA: Get correct overrides depending on dominant hand */
+    var animationData = {};
+    function getAnimationData() {
+        if (dominantHand === "right") {
+            animationData.rightHandType = 0;
+            animationData.isRightHandGrasp = false;
+            animationData.isRightIndexPoint = true;
+            animationData.isRightThumbRaise = false;
+            animationData.isRightIndexPointAndThumbRaise = false;
+        } else {
+            animationData.leftHandType = 0;
+            animationData.isLeftHandGrasp = false;
+            animationData.isLeftIndexPoint = true;
+            animationData.isLeftThumbRaise = false;
+            animationData.isLeftIndexPointAndThumbRaise = false;
+        }
+        return animationData;
+    }
+
+    /* SET UP HMD MODE: create controller mapping to listen for trigger presses */
+    var animationHandlerID;
+    function setUpHMDMode() {
+        if (controllerMapping) {
+            controllerMapping.enable();
+        }
+        animationHandlerID = MyAvatar.addAnimationStateHandler(getAnimationData, []);
+        Messages.sendLocalMessage("Hifi-Hand-Disabler", dominantHand);
+    }
+
+    /* SET UP DESKTOP MODE: Listen for mouse presses */
+    var mouseEventsConnected = false;
+    function setUpDesktopMode() {
+        if (!mouseEventsConnected) {
+            mouseEventsConnected = true;
+            Controller.mousePressEvent.connect(mousePressed);
+            Controller.mouseMoveEvent.connect(mouseContinueLine);
+            Controller.mouseReleaseEvent.connect(mouseReleased);
+        }
+    }
+
+    /* CLOSE HMD MODE: Remove controller mapping */
+    function closeHMDMode() {
+        if (controllerMapping) {
+            controllerMapping.disable();
+        }
+        Messages.sendLocalMessage("Hifi-Hand-Disabler", "none");
+        if (animationHandlerID) {
+            animationHandlerID = MyAvatar.removeAnimationStateHandler(animationHandlerID);
+        }
+    }
+
+    /* CLOSE DESKTOP MODE: Stop listening for mouse presses */
+    function closeDesktopMode() {
+        if (mouseEventsConnected) {
+            mouseEventsConnected = false;
+            Controller.mousePressEvent.disconnect(mousePressed);
+            Controller.mouseMoveEvent.disconnect(mouseContinueLine);
+            Controller.mouseReleaseEvent.disconnect(mouseReleased);
+        }
+    }
+
+    /* WHEN USER DOMAIN CHANGES: Close app to remove paint sphere in hand when leaving the domain */
+    var WAIT_TO_CLEAN_UP = 2000;
+    function domainChanged() {
+        Script.setTimeout(function() {
+            cleanUp();
+        }, WAIT_TO_CLEAN_UP);
+    }
+
+    /* WHEN USER CHANGES DOMINANT HAND: Switch default hand to place paint sphere in */
+    var WAIT_TO_REOPEN_APP_MS = 500;
+    function handChanged() {
+        if (MyAvatar.getDominantHand() === dominantHand) {
```

Comment:
you could add a check for `paintSphere` in here too, and return if it's truthy. you don't need to do anything in here if the app is disabled, right?